holmes/core/tool_calling_llm.py

```
@@ -241,136 +245,212 @@ def call(  # type: ignore
-                        investigation=raw_response,
-                        user_prompt=post_process_prompt,
+                    perf_timing.measure("llm.completion")
+                # catch a known error that occurs with Azure and replace the error message with something more obvious to the user
+                except BadRequestError as e:
+                    if (
+                        "Unrecognized request arguments supplied: tool_choice, tools"
+                        in str(e)
+                    ):
+                        raise Exception(
+                            "The Azure model you chose is not supported. Model version 1106 and higher required."
+                        )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)